### DISCUSSION TITLE
page.type loses focus?

### DISCUSSION THREAD

**Discussion by galaczi** [ZFC-Digital/puppeteer-real-browser#235] [Q&A] [answered]
Has anyone faced an issue that ``page.type`` loses focus while typing into an input? It is an Angular website that does input validation while typing. During manual testing the focus is not lost. Adding delay makes it worse.

**Comment by RubberArchind** [comment, marked as answer]
i have experience the same thing on some website, my workaround was to do this

```
await page.focus('textarea#text');

await page.keyboard.type(msg);
```

**Reply by galaczi** [reply]
I ended up writing a function that checks the value of the input after typing is finished, with 3 retries in case it doesn't match my intended input.

**Reply by RubberArchind** [reply]
is the website using recaptcha v3? 

if not I think it's fine to do something like page.evaluate then directly set the value via JavaScript (element.value = value)

**Reply by galaczi** [reply]
It's turnstile. I have not tested it with page.evaluate, might work.
